src/utils/Cookie.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+/*
+ * This file is part of RakLib.
+ * Copyright (C) 2014-2022 PocketMine Team <https://github.com/pmmp/RakLib>
+ *
+ * RakLib is not affiliated with Jenkins Software LLC nor RakNet.
+ *
+ * RakLib is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ */
+
+declare(strict_types=1);
+
+namespace raklib\utils;
+
+use raklib\utils\InternetAddress;
+use pocketmine\utils\Binary;
+use function random_int;
+use function crc32;
+
+final class Cookie{
+
+	/**
+	 * @var array<string, int> $cookies
+	 */
+	private array $cookies = [];
```

Comment:
Will this lead to a memory leak, especially when DDoS spams many cookies from different source addresses?

Reply:
if the cookie I send in OpenConnReply1 and the cookie I receive in OpenConnReq2 do not match, it should be discarded from the server.
Whether the cookie data matches or not, once it checks it, it immediately deletes it from the array.

Reply:
You will never unset it if the client never sends OpenConnReq2